From a0dd4da76edfffe975dedcea0d22ccefd8a4fa14 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Wed, 12 Jun 2024 06:46:21 +0200
Subject: [PATCH] [PATCH] Make sure generated service files get a .service
 suffix

ls2 expects all files to end with a proper .service suffix so we have to warant that when
generating them on the fly.

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
Upstream-Status: Pending
---
 serviceinstall.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/serviceinstall.cpp b/serviceinstall.cpp
index 8525f98..84e98f9 100644
--- a/serviceinstall.cpp
+++ b/serviceinstall.cpp
@@ -80,7 +80,7 @@ static string getPrivateEndpointDirectory(string root)
 static void generateEndpoint(string id, string serviceDirectory, string endpointDirectory)
 {
 	string line;
-	string destinationPath = endpointDirectory + "/" + id;
+	string destinationPath = endpointDirectory + "/" + id + ".service";
 	DIR *d;
 	
 	if ((d=opendir(endpointDirectory.c_str())) == NULL) {
@@ -102,7 +102,7 @@ static void generateEndpoint(string id, string serviceDirectory, string endpoint
 }
 static void deleteEndpoint(string id, string endpointDirectory)
 {
-	string destinationPath = endpointDirectory + "/" + id;
+	string destinationPath = endpointDirectory + "/" + id + ".service";
 	if (fexists(destinationPath)) {
 		DBG("Removing %s", destinationPath.c_str());
 		(void)unlink(destinationPath.c_str());
